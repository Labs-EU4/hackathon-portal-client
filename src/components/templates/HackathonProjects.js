import React, { useState, useEffect } from "react";
import { useParams } from 'react-router-dom';
import { useSelector } from "react-redux";
import Rating from "react-rating";

import {
  StyledWideBody,
  StyledCardWide,
  StyledParagraph,
  StyledRowBody,
  Team,
  SubmissionEntry,
  SubmissionContent,
  Strong,
  Description,
  RatingGroup,
  JudgeCount
} from '../../assets/styles/templates/HackathonProjects';
import emptyStar from "../../assets/images/star-hollow.png";
import fullStar from "../../assets/images/star-full.png";
import { H3 } from "../../assets/styles/atoms/HeadingStyling";
import { RowHead } from "../../assets/styles/atoms/RowHeadStyling";
import { Paragraph } from "../../assets/styles/atoms/ParagraphStyling";
import Button from "../atoms/Button";
import Spinner from "../molecules/Spinner";
import HackathonProjectPage from '../views/HackathonProjectPage';
import { useSubmissions } from "../../hooks";

const HackathonProjects = (
  // { id, setIsSubmissionsPageOpen }
) => {
  // const [ isProjectPageOpen, setIsProjectPageOpen ] = useState(false);
  // const [ projectId, setProjectId ] = useState(null);
  const { id } = useParams();
  const { event_title } = useSelector(state =>
    state.events.data.find(event => event.id === Number(id))
  );
  const [submissions, fetchSubmissions, loading] = useSubmissions(id);

  useEffect(() => {
    fetchSubmissions();
  }, [fetchSubmissions]);

  // const viewProjectHandler = (submissionId) => {
  //   setProjectId(submissionId);
  //   setIsProjectPageOpen(true);
  // }

  const renderSubmission = (s) => {
    return (
      <SubmissionEntry key={s.id}>
        <Team>{s.participant_or_team_name || s.project_title}</Team>
        <SubmissionContent>
          <Description>{s.project_writeups}</Description>
          {s.average_rating > 0 ? (
            <RatingGroup>
              <Rating
                initialRating={s.average_rating}
                readonly
                emptySymbol={
                  <img alt="Rubric star" src={emptyStar} />
                }
                fullSymbol={
                  <img alt="Rubric star" src={fullStar} />
                }
              />
              <JudgeCount>
                {`${s.acted_judges}/${s.number_of_judges +
                  1} voted`}
              </JudgeCount>
            </RatingGroup>
          ) : (
            <Paragraph strong>Not rated</Paragraph>
          )}
          {/* <Button
            color="blue"
            onClick={() => viewProjectHandler(s.id)}
          > */}
          <Button
            link
            color="blue"
            to={`/dashboard/event/${id}/project/${s.id}`}
          >View Project</Button>
        </SubmissionContent>
      </SubmissionEntry>
    );
  };

  return (
    <StyledWideBody>
      <StyledCardWide>
        <RowHead>
          <H3>
            Submitted projects for <Strong>"{event_title}"</Strong>
          </H3>
        </RowHead>
        {loading ? (
          <Spinner />
        ) : (
          <>
            <StyledRowBody>
              {submissions.length === 0 && (
                <StyledParagraph strong>No projects were submitted for this hackathon</StyledParagraph>
              )}
              {submissions.map(s => renderSubmission(s))}
            </StyledRowBody>
            {/* <Button onClick={() => setIsSubmissionsPageOpen(false)} color="grey">
              Back to event
            </Button> */}
            <Button link to={`/dashboard/event/${id}`} color="grey">
                Back to event
            </Button>
          </>
        )}
      </StyledCardWide>
      {/* {
        isProjectPageOpen && <HackathonProjectPage {...{id}} {...{projectId}} {...{setIsProjectPageOpen}}/>
      } */}
    </StyledWideBody>
  );
};

export default HackathonProjects;









// import React, { useEffect } from "react";
// import { useParams } from "react-router-dom";
// import { useSelector } from "react-redux";
// import {
//   BodyContainerColumn,
//   Card,
//   Team,
//   Strong,
//   Description,
//   RatingGroup,
//   JudgeCount,
//   SubmissionEntry
// } from "../styles/templates/HackathonProjectsStyling";
// import UserHeader from "../organisms/UserHeader";
// import { Footer } from "../organisms/index";
// import WideBody from "../atoms/WideBody";
// import Nav from "../molecules/Nav";
// import { H3 } from "../atoms/Heading";
// import { RowHead } from "../atoms/RowHead";
// import { RowBody } from "../atoms/RowBody";
// import { Column } from "../atoms/Column";
// import Button from "../atoms/Button";
// import Rating from "react-rating";
// import emptyStar from "../../assets/images/star-hollow.png";
// import fullStar from "../../assets/images/star-full.png";
// import { useSubmissions } from "../../hooks";
// import Spinner from "../molecules/Spinner";

// const HackathonProjects = () => {
//   const { id } = useParams();

//   const { event_title } = useSelector(state =>
//     state.events.data.find(event => event.id === Number(id))
//   );
//   const [submissions, fetchSubmissions, loading] = useSubmissions(id);

//   useEffect(() => {
//     fetchSubmissions();
//   }, [fetchSubmissions]);

//   return (
//     <div>
//       <UserHeader />
//       <WideBody>
//         <Nav />
//         <BodyContainerColumn>
//           <RowHead>
//             <H3>
//               Submitted projects for <Strong>&quot;{event_title}&quot;</Strong>
//             </H3>
//           </RowHead>

//           <Column>
//             <Card>
//               {loading ? (
//                 <Spinner />
//               ) : (
//                   <>
//                     <RowBody>
//                       {submissions.length === 0 &&
//                         "No projects were submitted for this hackathon."}
//                       {submissions.map((s, i) => {
//                         return (
//                           <SubmissionEntry key={s.id}>
//                             <Team>
//                               <H3>
//                                 {s.participant_or_team_name || s.project_title}
//                               </H3>
//                             </Team>
//                             <Description>{s.project_writeups}</Description>
//                             {s.average_rating > 0 ? (
//                               <RatingGroup>
//                                 <Rating
//                                   initialRating={s.average_rating}
//                                   readonly
//                                   emptySymbol={
//                                     <img alt="Rubric star" src={emptyStar} />
//                                   }
//                                   fullSymbol={
//                                     <img alt="Rubric star" src={fullStar} />
//                                   }
//                                 />
//                                 <JudgeCount>
//                                   {`${s.acted_judges}/${s.number_of_judges +
//                                     1} voted`}
//                                 </JudgeCount>
//                               </RatingGroup>
//                             ) : (
//                                 "Not rated."
//                               )}
//                             <Button
//                               anchor
//                               color="blue"
//                               to={`/dashboard/event/${id}/project/${s.id}`}
//                             >
//                               View
//                           </Button>
//                           </SubmissionEntry>
//                         );
//                       })}
//                     </RowBody>
//                     <Button anchor to={`/dashboard/event/${id}`} color="grey">
//                       Back to event
//                   </Button>
//                   </>
//                 )}
//             </Card>
//           </Column>
//         </BodyContainerColumn>
//       </WideBody>
//       <Footer />
//     </div>
//   );
// };

// export default HackathonProjects;
