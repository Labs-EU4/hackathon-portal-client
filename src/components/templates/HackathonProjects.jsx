import React, { useState, useEffect } from "react";
import { useParams, useLocation } from "react-router-dom";
import { useSelector } from "react-redux";
import Rating from "react-rating";

import {
  StyledWideBody,
  StyledCardWide,
  StyledParagraph,
  StyledRowBody,
  Team,
  SubmissionEntry,
  SubmissionContent,
  Strong,
  Description,
  RatingGroup,
  JudgeCount
} from "../../assets/styles/templates/HackathonProjectsStyling";
import emptyStar from "../../assets/images/star-hollow.png";
import fullStar from "../../assets/images/star-full.png";
import { H3 } from "../../assets/styles/atoms/HeadingStyling";
import { RowHead } from "../../assets/styles/atoms/RowHeadStyling";
import { Paragraph } from "../../assets/styles/atoms/ParagraphStyling";
import Button from "../atoms/Button";
import Spinner from "../molecules/Spinner";
import HackathonProjectPage from "../views/HackathonProjectPage";
import { useSubmissions } from "../../hooks";

const HackathonProjects = ({ setIsSubmissionsPageOpen }) => {
  const { id } = useParams();
  const [isProjectPageOpen, setIsProjectPageOpen] = useState(false);
  const [projectId, setProjectId] = useState(null);
  const { pathname } = useLocation();
  const currentPath = pathname.split("/")[1];
  const { event_title } = useSelector(state =>
    state.events.data.find(event => event.id === Number(id))
  );
  const [submissions, fetchSubmissions, loading] = useSubmissions(id);

  useEffect(() => {
    fetchSubmissions();
  }, [fetchSubmissions]);

  const viewProjectHandler = submissionId => {
    setProjectId(submissionId);
    setIsProjectPageOpen(true);
  };

  const renderSubmission = s => {
    return (
      <SubmissionEntry key={s.id}>
        <Team>{s.participant_or_team_name || s.project_title}</Team>
        <SubmissionContent>
          <Description>{s.project_writeups}</Description>
          {s.average_rating > 0 ? (
            <RatingGroup>
              <Rating
                initialRating={s.average_rating}
                readonly
                emptySymbol={<img alt="Rubric star" src={emptyStar} />}
                fullSymbol={<img alt="Rubric star" src={fullStar} />}
              />
              <JudgeCount>
                {`${s.acted_judges}/${s.number_of_judges + 1} voted`}
              </JudgeCount>
            </RatingGroup>
          ) : (
            <Paragraph strong>Not rated</Paragraph>
          )}
          <Button
            // link
            color="blue"
            // to={`/${currentPath}/event/${id}/project/${s.id}`}
            onClick={() => viewProjectHandler(s.id)}
          >
            View Project
          </Button>
        </SubmissionContent>
      </SubmissionEntry>
    );
  };

  return (
    <StyledWideBody>
      <StyledCardWide>
        <RowHead>
          <H3>
            Submitted projects for <Strong>"{event_title}"</Strong>
          </H3>
        </RowHead>
        {loading ? (
          <Spinner />
        ) : (
          <>
            <StyledRowBody>
              {submissions.length === 0 && (
                <StyledParagraph strong>
                  No projects were submitted for this hackathon
                </StyledParagraph>
              )}
              {submissions.map(s => renderSubmission(s))}
            </StyledRowBody>
            <Button link to={`/${currentPath}/event/${id}`} color="grey">
              Back to event
            </Button>
          </>
        )}
      </StyledCardWide>
      {isProjectPageOpen && (
        <HackathonProjectPage
          {...{ id }}
          {...{ projectId }}
          {...{ setIsProjectPageOpen }}
        />
      )}
    </StyledWideBody>
  );
};

export default HackathonProjects;
