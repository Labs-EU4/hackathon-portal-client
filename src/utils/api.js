import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
