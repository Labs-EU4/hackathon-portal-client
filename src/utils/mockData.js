export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
