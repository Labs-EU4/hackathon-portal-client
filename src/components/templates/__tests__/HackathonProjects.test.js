import React from "react";
import { Router, Route } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import { initialState } from "../../../utils/mockData";
import HackathonProjects from "../HackathonProjects";

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);

  jestFeatures = render(
    <Router history={history}>
      <Route
        match={{
          path: `/dashboard/event/:id/projects`,
          url: "/dashboard/event/1/projects",
          isExact: true,
          params: { id: 1 }
        }}
      >
        <Provider store={store}>
          <HackathonProjects />
        </Provider>
      </Route>
    </Router>
  );
});

describe("Component HackathonProjects.js renders properly", () => {
  it("asserts that the component renders properly", () => {
    console.log("THE COMPONENT IS NOT BEING LOGGED", jestFeatures);

    expect(jestFeatures.debug()).toMatchSnapshot();
  });
  //   it("the Image within the component is rendering properly", () => {
  //     expect(
  //       jestFeatures.getByAltText("Hackton - Organise hackathons")
  //     ).toBeInTheDocument();
  //   });
  //   it("the text node Log In is rendering properly", () => {
  //     expect(jestFeatures.queryByText(/Log In/i)).toBeInTheDocument();
  //   });
  //   it("the text node Sign Up is rendering properly", () => {
  //     expect(jestFeatures.queryByText(/Sign Up/i)).toBeInTheDocument();
  //   });
});
