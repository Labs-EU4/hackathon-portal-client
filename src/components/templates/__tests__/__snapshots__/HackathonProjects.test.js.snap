// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HackathonProjects.js renders properly asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-eHgmQL bylZnf"
        >
          <div
            class="sc-jWBwVP bYNjAt"
          >
            <a
              href="/dashboard"
            >
              <img
                alt="Hackton - Organise hackathons"
                class="sc-cMljjf jVjzeq"
                src="Hackton-logo.png"
              />
            </a>
            <div
              class="sc-kPVwWT MVsVr"
            >
              <div
                class="sc-iRbamj jYiCzw"
              >
                <span>
                  <svg>
                    link-burger-icon.svg
                  </svg>
                  Menu
                </span>
                <ul>
                  <li>
                    <a
                      class="sc-jlyJG cigUwh"
                      href="/dashboard"
                    >
                      Dashboard
                    </a>
                  </li>
                  <li>
                    <a
                      class="sc-jlyJG cigUwh"
                      href="/dashboard/profile"
                    >
                      Profile
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="sc-fMiknA iCUXtn"
              >
                8
                <ul
                  class="sc-dVhcbM hRRZmt"
                >
                  <li>
                    <a
                      href="/dashboard/profile/edit"
                    >
                      Settings
                    </a>
                  </li>
                  <li>
                    <span>
                      LogOut
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jtRfpW eEMFuT"
        >
          <div
            class="sc-gPEVay gwbcbn"
          >
            <a
              class="sc-jlyJG cigUwh"
              href="/dashboard"
            >
              <svg>
                link-dashboard-icon.svg
              </svg>
              Dashboard
            </a>
            <a
              class="sc-jlyJG cigUwh"
              href="/dashboard/profile"
            >
              <svg>
                link-profile-icon.svg
              </svg>
              Profile
            </a>
          </div>
          <div
            class="sc-bdVaJa cdzpxt sc-VigVT kLsUCd"
          >
            <div
              class="sc-gxMtzJ hscQJP"
            >
              <h3
                class="sc-jqCOkK jMeitG"
              >
                Submitted projects for 
                <strong
                  class="sc-jzJRlG irjRNl"
                >
                  "
                  EuroHack
                  "
                </strong>
              </h3>
            </div>
            <div
              class="sc-gzOgki fmasAF"
            >
              <div
                class="sc-htpNat sc-bxivhb sc-jTzLTM bLoYWy"
              >
                <div
                  class="sc-gZMcBi iOlOKS"
                  spacing="center"
                >
                  <div
                    aria-busy="true"
                    class=""
                  >
                    <svg
                      aria-label="audio-loading"
                      height="100"
                      stroke="#2559a1"
                      viewBox="0 0 44 44"
                      width="100"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fill-rule="evenodd"
                        stroke-width="2"
                      >
                        <circle
                          cx="22"
                          cy="22"
                          r="1"
                        >
                          <animate
                            attributeName="r"
                            begin="0s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.165, 0.84, 0.44, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 20"
                          />
                          <animate
                            attributeName="strokeOpacity"
                            begin="0s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.3, 0.61, 0.355, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 0"
                          />
                        </circle>
                        <circle
                          cx="22"
                          cy="22"
                          r="1"
                        >
                          <animate
                            attributeName="r"
                            begin="-0.9s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.165, 0.84, 0.44, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 20"
                          />
                          <animate
                            attributeName="strokeOpacity"
                            begin="-0.9s"
                            calcMode="spline"
                            dur="1.8s"
                            keySplines="0.3, 0.61, 0.355, 1"
                            keyTimes="0; 1"
                            repeatCount="indefinite"
                            values="1; 0"
                          />
                        </circle>
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-eqIVtm bOkwpW"
        >
          <div
            class="sc-caSCKo iHwPyD"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-kjoXOD cykQaQ"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-kjoXOD cykQaQ"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-kjoXOD cykQaQ"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-cHGsZl sc-kgAjT hVXAbA"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-cHGsZl sc-kgAjT hVXAbA"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-cHGsZl sc-kgAjT hVXAbA"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-htoDjs sc-dnqmqq xAsmd"
              >
                International Crafters Â© 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-eHgmQL bylZnf"
      >
        <div
          class="sc-jWBwVP bYNjAt"
        >
          <a
            href="/dashboard"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-cMljjf jVjzeq"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-kPVwWT MVsVr"
          >
            <div
              class="sc-iRbamj jYiCzw"
            >
              <span>
                <svg>
                  link-burger-icon.svg
                </svg>
                Menu
              </span>
              <ul>
                <li>
                  <a
                    class="sc-jlyJG cigUwh"
                    href="/dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <li>
                  <a
                    class="sc-jlyJG cigUwh"
                    href="/dashboard/profile"
                  >
                    Profile
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="sc-fMiknA iCUXtn"
            >
              8
              <ul
                class="sc-dVhcbM hRRZmt"
              >
                <li>
                  <a
                    href="/dashboard/profile/edit"
                  >
                    Settings
                  </a>
                </li>
                <li>
                  <span>
                    LogOut
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jtRfpW eEMFuT"
      >
        <div
          class="sc-gPEVay gwbcbn"
        >
          <a
            class="sc-jlyJG cigUwh"
            href="/dashboard"
          >
            <svg>
              link-dashboard-icon.svg
            </svg>
            Dashboard
          </a>
          <a
            class="sc-jlyJG cigUwh"
            href="/dashboard/profile"
          >
            <svg>
              link-profile-icon.svg
            </svg>
            Profile
          </a>
        </div>
        <div
          class="sc-bdVaJa cdzpxt sc-VigVT kLsUCd"
        >
          <div
            class="sc-gxMtzJ hscQJP"
          >
            <h3
              class="sc-jqCOkK jMeitG"
            >
              Submitted projects for 
              <strong
                class="sc-jzJRlG irjRNl"
              >
                "
                EuroHack
                "
              </strong>
            </h3>
          </div>
          <div
            class="sc-gzOgki fmasAF"
          >
            <div
              class="sc-htpNat sc-bxivhb sc-jTzLTM bLoYWy"
            >
              <div
                class="sc-gZMcBi iOlOKS"
                spacing="center"
              >
                <div
                  aria-busy="true"
                  class=""
                >
                  <svg
                    aria-label="audio-loading"
                    height="100"
                    stroke="#2559a1"
                    viewBox="0 0 44 44"
                    width="100"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke-width="2"
                    >
                      <circle
                        cx="22"
                        cy="22"
                        r="1"
                      >
                        <animate
                          attributeName="r"
                          begin="0s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.165, 0.84, 0.44, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 20"
                        />
                        <animate
                          attributeName="strokeOpacity"
                          begin="0s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.3, 0.61, 0.355, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 0"
                        />
                      </circle>
                      <circle
                        cx="22"
                        cy="22"
                        r="1"
                      >
                        <animate
                          attributeName="r"
                          begin="-0.9s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.165, 0.84, 0.44, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 20"
                        />
                        <animate
                          attributeName="strokeOpacity"
                          begin="-0.9s"
                          calcMode="spline"
                          dur="1.8s"
                          keySplines="0.3, 0.61, 0.355, 1"
                          keyTimes="0; 1"
                          repeatCount="indefinite"
                          values="1; 0"
                        />
                      </circle>
                    </g>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-eqIVtm bOkwpW"
      >
        <div
          class="sc-caSCKo iHwPyD"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-kjoXOD cykQaQ"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-kjoXOD cykQaQ"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-kjoXOD cykQaQ"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-cHGsZl sc-kgAjT hVXAbA"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-cHGsZl sc-kgAjT hVXAbA"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-cHGsZl sc-kgAjT hVXAbA"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-htoDjs sc-dnqmqq xAsmd"
            >
              International Crafters Â© 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
